test/distributed/test_c10d_common.py

```diff
@@ -1116,6 +1116,32 @@ def _test_warn_not_in_group(self, backend):
             dist.barrier(group=group)
             dist.broadcast(x, src=0, group=group)
 
+    def _test_rank_membership(self, backend):
+        store = dist.FileStore(self.file_name, self.world_size)
+        dist.init_process_group(
+            backend,
+            world_size=self.world_size,
+            rank=self.rank,
+            store=store,
+        )
+        self.assertTrue(self.world_size > 1)
+
+        group = dist.new_group(ranks=[1])
+        self.assertEqual(dist.get_local_rank(group, 1), 0)
+        with self.assertRaisesRegex(RuntimeError, "not part of group"):
+            dist.get_local_rank(group, 0)
+        with self.assertRaisesRegex(RuntimeError, "not registered"):
+            dist.get_local_rank(DummyProcessGroup(self.rank, self.world_size), 0)
+
+        self.assertEqual(dist.get_global_rank(group, 0), 1)
```

Comment:
Hmm, this is honestly a bit confusing for me, because `get_local_rank` seems to take in a global rank, and `get_global_rank` seems to take in the group's local rank. 

Also, one question around the semantics / definition of "local rank". Does this mean the relative rank within the node? 

i.e. what if I initialize

```
# assume global ranks contain 1-16 across 2 nodes

group = dist.new_group(ranks=range(1, 16))
```

which spans devices across 2 nodes.

What would `dist.get_local_rank(group, 9)` return?

Reply:
The naming is confusing because some times local rank is used in reference to the host relative rank.
Perhaps it should be renamed to group rank?

It is confusing that `get_global_rank` takes a local rank and vice-versa, but that's the nature of this beast.

A clearer example:

```
pg = dist.new_group(ranks=[1, 5, 10, 20])

assert pg.get_size() == 4

# global rank 1 is local rank 0
# global rank 5 is local rank 1
# ...

assert dist.get_local_rank(pg, 5) == 1
assert dist.get_local_rank(pg, 20) == 3

assert dist.get_local_rank(pg, 5) # raises

# and the inverse
assert dist.get_global_rank(pg, 0) == 1
assert dist.get_global_rank(pg, 1) == 5
assert dist.get_global_rank(pg, 2) == 10
assert dist.get_global_rank(pg, 5) # raises
```